src/client/Microsoft.Identity.Client/ApiConfig/Executors/ManagedIdentityExecutor.cs

```diff
@@ -35,19 +38,48 @@ public ManagedIdentityExecutor(IServiceBundle serviceBundle, ManagedIdentityAppl
         {
             var requestContext = CreateRequestContextAndLogVersionInfo(commonParameters.CorrelationId, cancellationToken);
 
+            if (_managedIdentityApplication.KeyMaterialManager.CryptoKeyType != CryptoKeyType.None)
```

Comment:
Suggest moving this to [AcquireTokenForManagedIdentityParameterBuilder](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/gladjohn/slc_cae_pub_preview/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs)